oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java

```diff
@@ -489,6 +491,11 @@ boolean isConfiguredForDomainWideDelegation() {
     return serviceAccountUser != null && serviceAccountUser.length() > 0;
   }
 
+  private GenericData parseResponseAs(HttpResponse response) throws IOException {
+    GenericData genericData = response.parseAs(GenericData.class);
+    LoggingUtils.logGenericData(genericData, LOGGER, "Auth response payload.");
+    return genericData;
+  }
```

Comment:
Can we move this to LoggingUtils since this seems to be used in the other credentials?

Reply:
They are slightly different, but I will try to extract any useful shared logics into LoggingUtils.

Reply:
Can you highlight the difference? From what I see, the other Credentials also end up calling `GenericData responseData = response.parseAs(GenericData.class);` but this is moved to a function.

Might be missing something, but I don't see what is preventing this from going into LoggingUtils.